snappy_pipeline/workflows/adapter_trimming/__init__.py

```diff
@@ -0,0 +1,334 @@
+# -*- coding: utf-8 -*-
+"""Implementation of the ``adapter_trimming`` step
+
+The adapter_trimming step performs adapter & quality trimming of reads (DNA or RNA).
+The tools are highly configurable, and provide feedback of the success of the operation.
+
+==========
+Step Input
+==========
+
+Adapter trimming starts at the raw RNA-seq reads.  Thus, the input is very similar to one of
+:ref:`ngs_mapping step <step_ngs_mapping>`.
+
+See :ref:`ngs_mapping_step_input` for more information.
+
+===========
+Step Output
+===========
+
+Adapter trimming will be performed for all NGS libraries in all sample sheets.  For each combination
+of tool library, a directory ``{tool}.{lib_name}-{lib_pk}/out`` will be created.
+Therein, trimmed fastq files will be created, together with fastq files for reads that might be
+filtered out by the process.
+
+For example, it might look as follows for the example from above:
+
+::
+
+    output/
+    +-- bbduk.P001-N1-DNA1-WES1
+    |   `-- out
+    |       |-- Sample001_Lane_001_R1.fastq.gz
+    |       |-- Sample001_Lane_001_R2.fastq.gz
+    |       |-- Sample001_Lane_002_R1.fastq.gz
+    |       |-- Sample001_Lane_002_R2.fastq.gz
+    |       |-- Sample001_Lane_001_R1.fastq.gz.md5
+    |       |-- Sample001_Lane_001_R2.fastq.gz.md5
+    |       |-- Sample001_Lane_002_R1.fastq.gz.md5
+    |       |-- Sample001_Lane_002_R2.fastq.gz.md5
+    |       `-- .done
+    [...]
+
+=====================
+Default Configuration
+=====================
+
+The default configuration is as follows.
+
+.. include:: DEFAULT_CONFIG_adapter_trimming.rst
+
+================================
+Available Adapter Trimming Tools
+================================
+
+The following adpter trimming tools are currently available
+
+- ``"bbduk"``
+- ``"fastp"``
+
+"""
+
+from collections import OrderedDict
+import os
+
+from biomedsheets.shortcuts import GenericSampleSheet
+from snakemake.io import expand
+
+from snappy_pipeline.base import UnsupportedActionException
+from snappy_pipeline.utils import dictify, listify
+from snappy_pipeline.workflows.abstract import (
+    BaseStep,
+    BaseStepPart,
+    LinkInPathGenerator,
+    LinkInStep,
+    LinkOutStepPart,
+    ResourceUsage,
+    get_ngs_library_folder_name,
+)
+
+#: Adatper trimming tools
+TRIMMERS = ("bbduk", "fastp")
+
+#: Default configuration for the hla_typing schema
+DEFAULT_CONFIG = r"""
+# Default configuration ngs_mapping
+step_config:
+  adapter_trimming:
+    path_link_in: ""
```

Comment:
Should this be an option in the config? It won't use for anything as far as I understood. This is the workflow that creates the linkin path for the others.